projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -85,9 +83,6 @@ public function get_item_permissions_check( $request ) { // phpcs:ignore Generic
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
 	public function get_item( $request ) { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter, VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		if ( ! ( new Host() )->is_wpcom_platform() && get_option( 'wpcom_admin_interface' ) !== 'wp-admin' ) {
```

Comment:
Took a look at this coming back from AFK and I'm concerned whether this removal will break the Calypso menus for self-hosted sites, since this was independent of whether the Masterbar module was active.
@mmtr Can I please ask you to take a look and let us know the intended behaviour for self-hosted sites depending on the `wpcom_admin_interface` option? Does the corresponding site menu in Calypso display as expected atm?

Reply:
Good catch, I did some testing just now on 13.7 vs 13.8, testing a WordPress.com `/home/siteurl` URL for a Jurassic Ninja site, and the menu is different. Assuming the menu is expected to be accessed independently to the WP.com Toolbar (I likely made an assumption that without the WordPress.com toolbar link and without Calypsoification it wasn't needed), I'll wait to hear back first before creating a PR to add that back.

Reply:
@fgiannar is correct. For self-hosted sites, we always need to load this file. This is responsible for how the menu works in Calypso, and should work regardless of whether the toolbar module is enabled or not (which is one of the reasons why I didn't like the admin-menu functionality to live in the masterbar code).

FWIW, the `wpcom_admin_interface` option is irrelevant here, because there is no logic for self-hosted sites that writes that option (i.e. the `wpcom_admin_interface` option will always be empty for self-hosted sites).